test/parallel/test-internal-errors.js

```diff
@@ -284,3 +284,49 @@ assert.strictEqual(
 assert.strictEqual(
   errors.message('ERR_INVALID_ASYNC_ID', ['asyncId', undefined]),
   'Invalid asyncId value: undefined');
+
+
+// Test that `code` property is mutable and that changing it does not change the
+// name.
+{
+  const myError = new errors.Error('ERR_TLS_HANDSHAKE_TIMEOUT');
+  assert.strictEqual(myError.code, 'ERR_TLS_HANDSHAKE_TIMEOUT');
+  assert.strictEqual(myError.hasOwnProperty('code'), false);
+  assert.strictEqual(myError.hasOwnProperty('name'), false);
+  assert.deepStrictEqual(Object.keys(myError), []);
+  const initialName = myError.name;
+  myError.code = 'FHQWHGADS';
+  assert.strictEqual(myError.code, 'FHQWHGADS');
+  assert.strictEqual(myError.name, initialName);
+  assert.deepStrictEqual(Object.keys(myError), ['code']);
+  assert.ok(myError.name.includes('ERR_TLS_HANDSHAKE_TIMEOUT'));
```

Comment:
IMHO these two are redundant since L300 `assert.strictEqual(myError.name, initialName);` covers that.
I'd add:
```js
  myError.name = 'hgnsdhjsd';
  assert.strictEqual(myError.name , 'hgnsdhjsd');
  assert.deepStrictEqual(Object.keys(myError), ['code', 'name']);
  assert.deepStrictEqual(myError.toString(), 'hgnsdhjsd: FHQWHGADS');
  const eProto = Object.getPrototypeOf(myError);
  const getCode = Object.getOwnPropertyDescriptor(eProto, 'code').get;
  assert.deepStrictEqual(getCode.call(myError), 'ERR_TLS_HANDSHAKE_TIMEOUT');
  delete myError.code;
  delete myError.name;
  assert.deepStrictEqual(myError.toString().includes('Error:'), true);
  assert.deepStrictEqual(myError.toString().includes('ERR_TLS_HANDSHAKE_TIMEOUT'), true);
```